### PR TITLE
[PAG-287] add pagopa devops az projects

### PR DESCRIPTION
**Pull request by pasqualespica**
### Description

Create AZ devops project for [mockPA](https://pagopa.atlassian.net/wiki/spaces/PAG/pages/162168930/MockPA+OnCloud+-+AZURE) ec service., to deploy [mock.ec-services-test](https://github.com/pagopa/mock.ec-services-test) nodejs express app

### CODE REVIEW

File: azure-devops/projects/pagopa-projects/mock-ec-services.tf [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  mock-ec-services-variables_secret = {
+
+  }
+  # code_review vars
+  mock-ec-services-variables_code_review = {
+    sonarcloud_service_conn = "SONARCLOUD-SERVICE-CONN"
+    sonarcloud_org          = var.mock-ec-services.repository.organization
+    sonarcloud_project_key  = "${var.mock-ec-services.repository.organization}_${var.mock-ec-services.repository.name}"
+    sonarcloud_project_name = var.mock-ec-services.repository.name
+  }
```
Comment (pasqualedevita) [inline azure-devops/projects/pagopa-projects/mock-ec-services.tf:31]: do u have integrated sonar cloud?

File: azure-devops/projects/pagopa-projects/mock-ec-services.tf [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  )
+
+  service_connection_ids_authorization = [
+    azuredevops_serviceendpoint_github.io-azure-devops-github-ro.id,
+    local.azuredevops_serviceendpoint_sonarcloud_id,
+  ]
+}
+
+module "mock-ec-services_deploy" {
+  source = "git::https://github.com/pagopa/azuredevops-tf-modules.git//azuredevops_build_definition_deploy_cstar?ref=v0.0.5"
```
Comment (pasqualedevita) [inline azure-devops/projects/pagopa-projects/mock-ec-services.tf:73]:
```suggestion
  source = "git::https://github.com/pagopa/azuredevops-tf-modules.git//azuredevops_build_definition_deploy?ref=v0.0.5"
```

File: azure-devops/projects/pagopa-projects/template/template-pipeline-microservice.tf [hunk 3]
```diff
@@ -0,0 +1,100 @@
+variable "template-pipeline-microservice" {
```
Comment (pasqualedevita) [inline azure-devops/projects/pagopa-projects/template/template-pipeline-microservice.tf:1]: remove this file

File: azure-devops/projects/pagopa-projects/template/template-pipeline-package.tf [hunk 4]
```diff
@@ -0,0 +1,96 @@
+variable "template-pipeline-package" {
```
Comment (pasqualedevita) [inline azure-devops/projects/pagopa-projects/template/template-pipeline-package.tf:1]: remove this file

**Review by pasqualedevita — CHANGES_REQUESTED** [review]
Has pagopa-github-bot admin rights on `mock.ec-services-test`?

File: azure-devops/projects/pagopa-projects/mock-ec-services.tf [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+module "mock-ec-services_code_review" {
+  source = "git::https://github.com/pagopa/azuredevops-tf-modules.git//azuredevops_build_definition_code_review_cstar?ref=v0.0.5"
+  count  = var.mock-ec-services.pipeline.enable_code_review == true ? 1 : 0
+
+  project_id                   = azuredevops_project.project.id
+  repository                   = var.mock-ec-services.repository
+  github_service_connection_id = azuredevops_serviceendpoint_github.io-azure-devops-github-pr.id
+
+  pull_request_trigger_use_yaml = true
```
Comment (pasqualedevita) [inline azure-devops/projects/pagopa-projects/mock-ec-services.tf:54]:
```suggestion
```

File: azure-devops/projects/pagopa-projects/mock-ec-services.tf [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+
+module "mock-ec-services_deploy" {
+  source = "git::https://github.com/pagopa/azuredevops-tf-modules.git//azuredevops_build_definition_deploy_cstar?ref=v0.0.5"
+  count  = var.mock-ec-services.pipeline.enable_deploy == true ? 1 : 0
+
+  project_id                   = azuredevops_project.project.id
+  repository                   = var.mock-ec-services.repository
+  github_service_connection_id = azuredevops_serviceendpoint_github.io-azure-devops-github-pr.id
+
+  ci_trigger_use_yaml = true
```
Comment (pasqualedevita) [inline azure-devops/projects/pagopa-projects/mock-ec-services.tf:80]:
```suggestion
```

**Review by pasqualedevita — CHANGES_REQUESTED** [review]

File: azure-devops/projects/pagopa-projects/mock-ec-services.tf (hunk 1 above)
Comment (pasqualespica) [reply azure-devops/projects/pagopa-projects/mock-ec-services.tf:31]: > Has pagopa-github-bot admin rights on `mock.ec-services-test`?
No, you're right, i added it to now

Comment (pasqualespica) [reply azure-devops/projects/pagopa-projects/mock-ec-services.tf:31]: > do u have integrated sonar cloud?

No, i'll remove it

File: azure-devops/projects/pagopa-projects/template/template-pipeline-microservice.tf (hunk 3 above)
Comment (pasqualespica) [reply azure-devops/projects/pagopa-projects/template/template-pipeline-microservice.tf:1]: ok

File: azure-devops/projects/pagopa-projects/template/template-pipeline-package.tf (hunk 4 above)
Comment (pasqualespica) [reply azure-devops/projects/pagopa-projects/template/template-pipeline-package.tf:1]: ok

File: azure-devops/projects/pagopa-projects/time_sleep.tf [hunk 7]
```diff
@@ -0,0 +1,7 @@
+# This is to work around an issue with azuredevops_resource_authorization
```
[the remark this answers is not in the document]
Comment (pasqualespica) [reply azure-devops/projects/pagopa-projects/time_sleep.tf:1]: ok

File: azure-devops/projects/pagopa-projects/service_connections.tf [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+  depends_on = [azuredevops_project.project]
+
+  project_id                = azuredevops_project.project.id
+  service_endpoint_name     = "pagopa-azurecr-dev"
+  resource_group            = "pagopa-d-aks-rg"
+  azurecr_name              = "pagopadacr"
+  azurecr_subscription_name = "DEV-PAGOPA"
+  azurecr_spn_tenantid      = module.secrets.values["PAGOPAIT-TENANTID"].value
+  azurecr_subscription_id   = module.secrets.values["PAGOPAIT-DEV-PAGOPA-SUBSCRIPTION-ID"].value
+}
```
[the remark this answers is not in the document]
Comment (pasqualespica) [reply azure-devops/projects/pagopa-projects/service_connections.tf:69]: ok, tnx

**Review by pasqualedevita — APPROVED** [review]

**Pull request merged**